packages/git/src/node/dugite-git.ts

```diff
@@ -694,18 +698,28 @@ export class DugiteGit implements Git {
         return undefined;
     }
 
-    private async getRemotes(repositoryPath: string): Promise<string[]> {
+    private async getRemotes(repositoryPath: string): Promise<Remote[]> {
         await this.ready.promise;
         const [exec, env] = await Promise.all([this.execProvider.exec(), this.gitEnv.promise]);
-        const result = await git(['remote'], repositoryPath, 'remote', { exec, env });
+        const result = await git(['remote', '-v'], repositoryPath, 'remote', { exec, env });
         const out = result.stdout || '';
-        return out.trim().match(/\S+/g) || [];
+        const results = out.trim().match(/\S+/g);
+        if (results) {
+            const values: Remote[] = [];
+            for (let i = 0; i < results.length; i += 6) {
+                values.push({ name: results[i], fetch: results[i + 1], push: results[i + 4] });
```

Comment:
Small remark: if you send the remote URL as is, you will transfer it through the wire as a text. Note, access tokens are secret. I am not sure if I ever want to see it in the logs or the UI. What does VS Code do?

Reply:
Thanks @kittaakos I haven't thought about the case where `access tokens` are included in the url.
I wonder what the appropriate handling would be when it's the case (for logs, gui, and over the wire). I know that `vscode` does display this information but they probably won't encounter the same ' over the wire' issue since they probably won't communicate that way.

This issue leads me to believe we may currently have the same issue for other components such as transferring **Terminal** output over the wire but that is still to be investigated and confirmed.

If you don't believe the PR adds any additional value and risks too much regarding security I'm fine with scrapping it :) I noticed that `vscode` had additional info when displaying their remote items, so I thought it'd be a nice feature to include but not at the expense of risking security.

Reply:
I am fine with this PR, just wanted to make sure we are aware of this.

Reply:
I assume will need to investigate and find an appropriate way to handle the `access token` security risk before being able to proceed with the PR.